### PR TITLE
Convert lib/auth/server to typescript

### PR DESCRIPTION
**Pull request by Page-**
Depends on #1597

### CODE REVIEW

**Comment by Page-** [comment]
tested @pdcastro

File: lib/auth/server.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+const createServer = ({ port }: { port: number }) => {
+	const app = express();
+	app.use(
+		bodyParser.urlencoded({
+			extended: true,
+		}),
+	);
+
+	app.set('view engine', 'ejs');
+	app.set('views', path.join(__dirname, 'pages'));
```
[the remark this answers is not in the document]
Comment (Page-) [reply lib/auth/server.ts]: I hadn't considered that it referred to files that would also be unused, I'll look into deleting them as part of this as well

[the remark this answers is not in the document]
Comment (Page-) [reply lib/auth/server.ts]: I checked and both css and images are used by the views, they just in-line them in the html itself and so don't need/use the express.static module

[the remark this answers is not in the document]
Comment (pdcastro) [reply lib/auth/server.ts]: Oh right! 👍  I'm glad you noticed it before deleting the files. :-)

**Comment by pdcastro** [comment]
```
+ npm link
...
npm ERR! code EEXIST
npm ERR! path C:\Users\Administrator\AppData\Roaming\npm\node_modules\@balena.io\resinci-deploy\index.js
npm ERR! dest C:\Users\Administrator\AppData\Roaming\npm\resinci-deploy
npm ERR! EEXIST: file already exists, cmd shim 'C:\Users\Administrator\AppData\Roaming\npm\node_modules\@balena.io\resinci-deploy\index.js' -> 'C:\Users\Administrator\AppData\Roaming\npm\resinci-deploy'
npm ERR! File exists: C:\Users\Administrator\AppData\Roaming\npm\resinci-deploy
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.
```
The `npm link` command comes from this line: https://github.com/balena-io/balena-concourse/blob/v5.6.19/pipelines/github-events/template.yml#L1037

I can reproduce it by running `npm link` on a [fly powershell](https://concourse-ci.org/fly.html). I've also manually attempted `npm link --force`, which succeeds.

**Pull request merged**